File: app/src/main/java/com/github/fo2rist/mclaren/dagger/StoriesFragmentModule.java
package com.github.fo2rist.mclaren.dagger;

import com.github.fo2rist.mclaren.analytics.EventsLogger;
import com.github.fo2rist.mclaren.mvp.FeedContract;
import com.github.fo2rist.mclaren.repository.FeedRepository;
import com.github.fo2rist.mclaren.repository.FeedRepositoryEventBus;
import com.github.fo2rist.mclaren.ui.feedscreen.FeedPresenter;
import dagger.Module;
import dagger.Provides;
import javax.inject.Named;

@Module
class StoriesFragmentModule {
    @Provides
    @Scopes.PerFragment
    FeedContract.Presenter providePresenter(
            @Named("storystream-feed") FeedRepository repository,
            FeedRepositoryEventBus eventBus,
            EventsLogger logger
    ) {
        return new FeedPresenter(repository, eventBus, logger);
    }
}

File: app/src/main/java/com/github/fo2rist/mclaren/dagger/StoriesFragmentModule.kt
package com.github.fo2rist.mclaren.dagger

import com.github.fo2rist.mclaren.mvp.FeedContract
import com.github.fo2rist.mclaren.repository.feed.FeedRepository
import com.github.fo2rist.mclaren.ui.feedscreen.FeedPresenter
import com.github.fo2rist.mclaren.ui.feedscreen.StoriesFeedFragment
import dagger.Binds
import dagger.Module
import javax.inject.Named

@Module
internal interface StoriesFragmentModule {

    @Binds
    @Scopes.PerFragment
    fun provideStoriesFeedView(view: StoriesFeedFragment): FeedContract.View

    @Binds
    @Scopes.PerFragment
    fun provideStoryStreamRepository(@Named("storystream-feed") repository: FeedRepository): FeedRepository

    @Binds
    @Scopes.PerFragment
    fun providePresenter(presenter: FeedPresenter): FeedContract.Presenter
}
